lektor_s3.py

```diff
@@ -142,14 +155,53 @@ def compute_diff(self, local, remote):
                 diff['delete'].append(k)
         return diff
 
+    def create_headers(self, filename):
+        """ Set file headers from mimetype and config. """
+        mime, _ = mimetypes.guess_type(filename)
+        if mime is None:
+            mime = "text/plain"
+        headers = {'ContentType': mime}
+
+        # Set defaults first
+        for upload_arg in TransferManager.ALLOWED_UPLOAD_ARGS:
+            upload_arg_value = self.config.defaults().get(upload_arg.lower())
+            if upload_arg_value:
+                headers.update({upload_arg: upload_arg_value})
+
+        # Set file-specific rules
+        config_sections = self.config.sections()
+        for section in config_sections:
+            items = self.config.items(section)
+            options = dict((x, y) for x, y in items)
+            section_applies = False
+
+            if 'match' in options:
+                match = self.config.get(section, 'match')
+                if re.search(match, filename):
+                    section_applies = True
+
+            if not section_applies and 'extensions' in options:
+                ext = filename.split('.')[-1]
```

Comment:
Sure. That includes the extension separator so going with `os.path.splitext(filename)[1][1:]`.